workers.py

```diff
@@ -115,5 +116,63 @@ def worker() -> None:
 
         logger.debug('%s: Done, exiting', thread.name)
 
-    thread = Thread(target=worker)
+    def worker_topology(topology_info: dict) -> None:
+        """Download and forward the content."""
+        thread = current_thread()
+        logger.debug('%s: Worker started', thread.name)
+
+        # Build the POST data object
+        data = {
+            'id': source_id,
+            'data': {}
+        }
+
+        endpoint = topology_info['authentication']['endpoint']
+        username = topology_info["authentication"]['username']
+        password = topology_info["authentication"]['password']
+
+        for entity in topology_info['queries'].keys():
+            prometheus_metrics.METRICS['gets'].inc()
+
+            query_string = topology_info['queries'][entity]
+            try:
+                resp = _retryable(
+                    'get',
+                    f'{endpoint}/{entity}{query_string}',
+                    auth=(username, password),
+                    verify=False
+                )
+                data['data'][entity] = resp.json()
+                prometheus_metrics.METRICS['get_successes'].inc()
+            except requests.HTTPError as exception:
+                prometheus_metrics.METRICS['get_errors'].inc()
+                logger.error(
+                    '%s: Unable to fetch source data for "%s": %s',
+                    thread.name, source_id, exception
+                )
+                return
+
+        # Pass to next service
+        prometheus_metrics.METRICS['posts'].inc()
+        try:
+            resp = _retryable('post', f'http://{dest_url}', json=data)
```

Comment:
Since I've merged the #19, you would need to pass the headers with `rh_identity` here as well.. :wink:

Reply:
Yes, I was waiting for #19 to be merged :)